metrics/report/report_dockerfile/tidy_scaling_nc.R

```diff
@@ -0,0 +1,155 @@
+			#    300   6   14   19
+			ltpt$testname=testname
+			latencydata=rbind(latencydata, ltpt)
+		}
+	}
+}
+
+# Visualize data.
+if (length(latencydata[[1]]) <= 5 || length(unique(latencydata$testname)) > 1) {
+	# If there are many tests to compare or only few data points, use boxplot with extra percentile points.
+	latp = ggplot(data=latencydata, aes(x=n_pods)) + ylab("Latency (us)") + xlab("pods") + scale_y_continuous(labels=comma)
+	perc_mid = floor((perc_count)/2)
+	# Create boxplot around the middle percentile
+	if (perc_count >= 3) {
+		box_bottom=names(ltpt)[perc_mid+1]
+		box_mid=names(ltpt)[perc_mid+2]
+		box_top=names(ltpt)[perc_mid+3]
+		if (perc_count >= 5) {
+			whis_low=names(ltpt)[perc_mid]
+			whis_high=names(ltpt)[perc_mid+4]
+			latp = latp + geom_boxplot(aes_string(group="interaction(testname,n_pods)",ymin=whis_low,lower=box_bottom,middle=box_mid,upper=box_top,ymax=whis_high,fill="testname"),stat="identity")
+		} else {
+			latp = latp + geom_boxplot(aes_string(group="interaction(testname,n_pods)",lower=box_bottom,middle=box_mid,upper=box_top,fill="testname"),stat="identity")
+		}
+	}
+	# Boxplot (above) covers at most 5 percentiles around the center (median).
+	# Visualize the rest using a point for each percentile.
+	if (perc_count > 5) {
+		for (n in seq(1, (perc_count-5)/2)) {
+			lower_name=names(ltpt)[n+1]
+			upper_name=names(ltpt)[perc_count-n+2]
+			latp = latp + geom_point(aes_string(group="interaction(testname,n_pods)",y=lower_name, color="testname"))
+			latp = latp + geom_point(aes_string(group="interaction(testname,n_pods)",y=upper_name, color="testname"))
+		}
+	}
+} else {
+	# Use colored areas and median lines when there are many ticks on X axis
+	latp = ggplot(data=latencydata, aes(x=n_pods)) + ylab("Latency (us)") + xlab("pods") + scale_y_continuous(labels=comma)
+	perc_mid = floor((perc_count)/2)
+	perc_maxdist = perc_mid
+	plot_number = 0
+	for (plot_test in unique(latencydata$testname)) {
+		plot_number = plot_number + 1
+		for (n in seq(perc_mid)) {
+			# First fill outmost areas, like p5..p25 and p75..p95,
+			# then areas closer to the middle, like p25..p50 and p50..p75
+			lower_name = names(ltpt)[n+1]
+			lower_next_name = names(ltpt)[n+2]
+			upper_name = names(ltpt)[perc_count-n+2]
+			upper_prev_name = names(ltpt)[perc_count-n+1]
+			alpha = 0.7 * ((n+1) / (perc_mid+1))**2
+			latp = latp + geom_ribbon(data=latencydata[latencydata$testname==plot_test,],aes_string(x="n_pods",ymin=lower_name,ymax=lower_next_name,fill="testname"),alpha=alpha)
+			latp = latp + geom_ribbon(data=latencydata[latencydata$testname==plot_test,],aes_string(x="n_pods",ymin=upper_prev_name,ymax=upper_name,fill="testname"),alpha=alpha)
+		}
+		median_index = match("p50", names(ltpt))
+		if (!is.na(median_index)) {
+			# Draw median line
+			latp = latp + geom_line(data=latencydata[latencydata$testname==plot_test,],aes_string(x="n_pods",y=names(ltpt)[median_index],color="testname"))
+		}
+	}
+}
+
+# Table presentation.
```

Comment:
I don't comprehend how the code relates to the final table - can you add some commentary please?
For instance, in my 20 pod run, I see a table in the pdf with rows for 1,5,10 pods - how is that math done? I thought maybe it would be 1,10,20 for a 20 pod run for instance?